drivers/lm75/lm75.c

```diff
@@ -0,0 +1,399 @@
+#define LM75_SHUTDOWN_MODE      0x01
+#define LM75_OS_POLARITY        0x04
+#define TMP1075_ONE_SHOT_MODE   0x81 /* also sets the shutdown register to 1 */
+
+#if IS_ACTIVE(MODULE_LM75A)
+    lm75_properties_t lm75a_properties = {
+        .os_res         = LM75A_OS_RES,
+        .os_mult        = LM75A_OS_MULT,
+        .temp_res       = LM75A_TEMP_RES,
+        .temp_mult      = LM75A_TEMP_MULT,
+        .os_shift       = LM75A_OS_SHIFT,
+        .temp_shift     = LM75A_TEMP_SHIFT,
+    };
+
+#elif IS_ACTIVE(MODULE_TMP1075)
+    const lm75_properties_t tmp1075_properties = {
+    .os_res         = TMP1075_OS_RES,
+    .os_mult        = TMP1075_OS_MULT,
+    .temp_res       = TMP1075_TEMP_RES,
+    .temp_mult      = TMP1075_TEMP_MULT,
+    .os_shift       = TMP1075_OS_SHIFT,
+    .temp_shift     = TMP1075_TEMP_SHIFT,
+    };
+
+#endif
+
+int lm75_init(lm75_t *dev, const lm75_params_t *params) {
+
+    dev->lm75_params = *params;
+    uint8_t config = (params->shutdown_mode) | (params->tm_mode << 1) \
+                   | (params->polarity << 2) | (params->fault_q << 3);
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the device ID register of the TMP1075 sensor to confirm it is a TMP1075 */
+    if (IS_USED(MODULE_TMP1075)) {
+        uint16_t deid = 0;
+        if (i2c_read_regs(I2C_BUS, I2C_ADDR, TMP1075_DEVICE_ID_REG, &deid, 2, 0) != 0) {
+            LOG_INFO("Error reading device ID\n");
+         }
+
+        deid = (uint16_t)ntohs(deid);
+        /* checks if the device ID corresponds to the TMP1075 sensor
+         * and extends the parameter configuration if so */
+        if (deid == 0x7500) {
+            puts("Device is a TMP1075");
+            config |= (params->conv_rate_reg << 5);
+        }
+        else {
+            puts("Device ID Register doesnt match");
+            return LM75_ERROR;
+        }
+    }
+    else if (IS_USED(MODULE_LM75A)) {
+        puts("Device is an LM75A");
+    }
+    else {
+        LOG_ERROR("Device not supported\n");
+    }
+
+    /* write the config byte into the configuration register */
+    if (i2c_write_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    return LM75_SUCCESS;
+}
+
+int lm75_get_temperature_raw(lm75_t *dev, int *temperature) {
+
+    int16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_TEMP_REG, &temp, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+    i2c_release(I2C_BUS);
+
+    /* since the value is read in little endian, it must be converted to big endian
+    * then it must be multiplied by its resolution
+    * and then shifted so only the correct number of bits is used */
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.res->temp_res \
```

Comment:
```suggestion
    *temperature = (int16_t)ntohs(temp) * dev->lm75_params.res->temp_res \
```

You are reading a short int from the device (network) and convert to host endianness -> so **N**etwork**toH**ost **s**hort